captum/optim/_utils/models.py

```diff
@@ -168,3 +172,93 @@ def collect_activations(
     catch_activ = ActivationFetcher(model, targets)
     activ_out = catch_activ(model_input)
     return activ_out
+
+
+class AvgPool2dLayer(torch.nn.Module):
+    """
+    AvgPool2d layer that also zeros padding of a specific value. This
+    layer is meant to be used to replace MaxPool2d layers.
+
+    Args:
+        kernel_size (int or tuple of int): The size of the window to
+            perform average pooling with.
+        stride (int or tuple of int, optional): The stride window size
+            to use.
+        padding (int or tuple of int): The amount of
+            zero padding to add to both sides.
+        ceil_mode (bool, optional): Whether to use ceil or floor for
+            creating the output shape.
+        value (Any): Used to return any padding that's meant to be ignored
+            by pooling layers back to zero.
+    """
+
+    def __init__(
+        self,
+        kernel_size: Union[int, Tuple[int, ...]],
+        stride: Optional[Union[int, Tuple[int, ...]]] = None,
+        padding: Union[int, Tuple[int, ...]] = 0,
+        ceil_mode: bool = False,
+        value: Optional[Any] = float("-inf"),
+    ) -> None:
+        super().__init__()
+        self.avgpool = torch.nn.AvgPool2d(
+            kernel_size=kernel_size,
+            stride=stride,
+            padding=padding,
+            ceil_mode=ceil_mode,
+        )
+        self.value = value
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.avgpool(x)
+        if self.value is not None:
+            x[x == self.value] = 0.0
+        return x
+
+
+def max2avg_pool2d(model, value: Optional[Any] = float("-inf")) -> None:
+    """
+    Replace all nonlinear MaxPool2d layers with their linear AvgPool2d equivalents.
+    This allows us to ignore nonlinear values when calculating expanded weights.
+
+    Args:
+        model (nn.Module): A PyTorch model instance.
+        value (Any): Used to return any padding that's meant to be ignored by
+            pooling layers back to zero.
+    """
+
+    for name, child in model._modules.items():
+        if isinstance(child, torch.nn.MaxPool2d):
+            new_layer = AvgPool2dLayer(
+                kernel_size=child.kernel_size,
+                stride=child.stride,
+                padding=child.padding,
+                ceil_mode=child.ceil_mode,
+                value=value,
+            )
+            setattr(model, name, new_layer)
+        elif child is not None:
+            max2avg_pool2d(child)
+
+
+class IgnoreLayer(torch.nn.Module):
+    """
+    This layer is made to take the place of nonlinear activation layers.
+    """
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return x
+
+
+def ignore_layer(model, layer) -> None:

```

Comment:
Yeah, both `replace_layers` and `skip_layer` are meant to help users in getting their model read for Captum in some way (adding redirected relu, removing nonlinear layers, etc...).

Reply:
Do we already have a `skip_layer` function somewhere else ? My suggestion was to rename `ignore_layer` to `skip_layer` because it might sounds better.

Reply:
@NarineK The `skip_layer` function was just the `replace_layers` with `SkipLayer` as the layer argument, so I removed it. I can re-add it if you think it's useful to keep so that users don't have to import both `replace_layers` and `SkipLayer`.

Reply:
Actually, I think it would be useful to keep, so I'll re-add it.

Reply:
@ProGamerGov, it's good to reduce the redundancy. I thought that Skip is a better synonym of Ignore. I didn't see that you originally removed a function called `skip_layer`.